src/util.ts

```diff
@@ -0,0 +1,37 @@
+import Generator from 'yeoman-generator';
+import chalk from 'chalk';
+
+/**
+ * Interactively prompt with generator questions or return configured answers.
+ *
+ * @param prompts
+ * @param uninteractive
+ * @param generator
+ */
+export async function promptOrUninteractive(
+  prompts: Generator.Questions,
+  uninteractive: boolean,
+  generator: Generator,
```

Comment:
I don't know how TS-friendly this is, but you I think you could drop the last two arguments and use this function like `promptOrUninteractive.apply(this, prompts)` inside the generators, and then you wouldn't need all the _promptOrUninteractive wrapper functions. Inside the body `this` would be the generator and you can access `this.options.uninteractive` directly.

Reply:
I managed to work this out by changing the function signature to this:

```javascript
export async function promptOrUninteractive(
  this: Generator,
  prompts: Generator.Questions,
): Promise<Generator.Answers> {
```

It seems that specifying the `this` parameter is handled uniquely to evaluate the runtime context rather than an argument to the function the way it looks. This allowed for all the improvements you described:

* Removed additional parameters from the `promptOrUninteractive()` function.
* Enforces proper execution of the method using `bind`/`call`/`apply` by type checking that a Generator is being bound.
* Allowed removal of all boilerplate `_promptOrUninteractive()` methods.
* Replaced all calls to the boilerplate methods with `promptOrUninteractive.call(this, prompts)`.

Thanks, @mrclay! This was an awesome recommendation and one I learned a fair bit from while implementing it!

Reply:
For reference, this refactor was applied in d4a2d5e.